File: src/node/magma/api/collect/Queries.ts
import { Query } from "../../../magma/api/collect/Query";
import { Option } from "../../../magma/api/option/Option";
import { HeadedQuery } from "../../../magma/api/collect/head/HeadedQuery";
import { EmptyHead } from "../../../magma/api/collect/head/EmptyHead";
import { Head } from "../../../magma/api/collect/head/Head";
import { SingleHead } from "../../../magma/api/collect/head/SingleHead";
export class Queries {
	static fromOption<T>(option: Option<T>): Query<T> {
		return new HeadedQuery<T>(option/*Option<T>*/.map((element: T) => Queries/*auto*/.getTSingleHead(element/*T*/)).orElseGet(() => new EmptyHead<T>(/*auto*/)));
	}
	static getTSingleHead<T>(element: T): Head<T> {
		return new SingleHead<T>(element/*T*/);
	}
	static fromArray<T>(elements: T[]): Query<T> {
		/*return new HeadedQuery<Integer>(new RangeHead(elements.length)).map((Integer index) -> elements[index])*/;
	}
}

File: src/java/magma/api/collect/Queries.java
package magma.api.collect;

import magma.api.collect.head.EmptyHead;
import magma.api.collect.head.Head;
import magma.api.collect.head.HeadedQuery;
import magma.api.collect.head.SingleHead;
import magma.api.option.Option;

public final class Queries {
    public static <T> Query<T> fromOption(final Option<T> option) {
        return new HeadedQuery<T>(option.map((T element) -> Queries.getTSingleHead(element)).orElseGet(() -> new EmptyHead<T>()));
    }

    private static <T> Head<T> getTSingleHead(final T element) {
        return new SingleHead<T>(element);
    }
}
